ngraph_bridge/ngraph_encapsulate_op.h

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright 2017-2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#ifndef NGRAPH_TF_ENCAPSULATE_OP_H_
+#define NGRAPH_TF_ENCAPSULATE_OP_H_
+
+#include <ostream>
+#include <vector>
+
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/graph/graph.h"
+
+#include "logging/ngraph_log.h"
+#include "ngraph/ngraph.hpp"
+#include "ngraph_bridge/ngraph_encapsulate_impl.h"
+#include "ngraph_bridge/ngraph_freshness_tracker.h"
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+class NGraphEncapsulateOp : public OpKernel {
+ public:
+  explicit NGraphEncapsulateOp(OpKernelConstruction* ctx);
+  ~NGraphEncapsulateOp() override;
+  void Compute(OpKernelContext* ctx) override;
+
+ private:
+  NGraphEncapsulateImpl* ng_encap_impl;
```

Comment:
You can create the object on the stack - why use a pointer? Also, if you must - then use a unique_ptr so that you don't need to remember to delete the pointer at the end. 
Also, since this is a data member use `m_` or `abc_` convention (follow whatever is used earlier in this file.